invenio_records_marc21/ui/theme/assets/semantic-ui/js/invenio_records_marc21/deposit/RecordSchema.js

```diff
@@ -0,0 +1,144 @@
+// This file is part of Invenio.
+//
+// Copyright (C) 2021 Graz University of Technology.
+//
+// Invenio-Records-Marc21 is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see LICENSE file for more
+// details.
+
+import _get from "lodash/get";
+import _has from "lodash/has";
+import axios from 'axios';
+
+export class RecordSchema {
+
+  /**
+   *  Make sure RecordSchema called first time with props.link
+   */
+
+	constructor(link, schema) {
+      this.schema = {}
+      this.link = link;
+      this.leader_field = "LDR";
+      if (schema !== {}){
+        this.setSchema(schema);
+        this.loaded = true;
+      } else {
+        this.loaded = false;
+      }
+	}
+
+  isReady(){
+    return this.loaded;
+  }
+
+  setSchema(schema){
+    this.schema = schema;
+  }
+
+  loadSchema() {
+    axios.get(this.state.link).then(
+      result => {
+        this.loaded =  true;
+        this.schema = result.data;
+      },
+      // Note: it's important to handle errors here
+      // instead of a catch() block so that we don't swallow
+      // exceptions from actual bugs in components.
+      error => {
+        this.loaded =  false;
+        this.schema = {};
+        this.error = error;
+      }
+    );
+  }
+
+  isLeaderField(key){
+    let keys = this.getLeaderFieldKeys();
+    return keys.includes(key);
+  }
+
+  isDataField(key){
+    let keys = this.getDataFieldKeys();
+    return keys.includes(key);
+  }
+
+  hasLeaderKey(key) {
+    return _has(this.schema, ["fields", this.leader_field, "positions", key])
+  }
+
+  getLeaderFieldKeys() {
+
+    let fields = this.getLeaderFields();
+    let keys = []
+    for (const [key, value] of Object.entries(fields)) {
+      keys.push(key);
+    }
+    return keys;
+  }
+
+  getLeaderFields() {
+    return _get(this.schema, ["fields", this.leader_field, "positions"], {});
+  }
+
+  getLeaderField(key) {
+    const leaderfields = this.getLeaderFields();
+    return leaderfields[key];
+  }
+
+
+  getLeaderFieldOptions(key) {
+    const leaderfield = this.getLeaderField(key);
+    const leaderfield_codes = _get(leaderfield, ["codes"], {});
+    let codes = this.generateDropdownOptions(leaderfield_codes);
+    return codes;
+  }
+
+  getDataFields() {
+    return _get(this.schema, ["fields"], {});
+  }
+
+
+  getDataFieldKeys() {
+    let fields = this.getDataFields();
+    let keys = []
+    for (const [key, value] of Object.entries(fields)) {
+      keys.push(key);
+    }
+    return keys;
+  }
+
+  getDataField(key) {
+    const datafield = this.getDataFields();
+    return datafield[key];
+  }
+
+  generateDropdownOptions(fields){
+    let codes = []
```

Comment:
minor: codes could also be set to const. push does not changes the variable it is still an array